warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/WhindexIngest.java

```diff
@@ -0,0 +1,212 @@
+            for (String rule : whindexRules.split(RULE_DELIMITER)) {
+                String[] parts = rule.split(FIELDS_VALUES_DELIMITER);
+                String[] curFields = parts[0].split(FIELD_VALUE_DELIMITER);
+                String[] curValues = parts[1].split(FIELD_VALUE_DELIMITER);
+
+                // Add each field from the current rule into the whindexFields map as a k.
+                // Each rule entry will have all related values as a v.
+                for (String field : curFields) {
+
+                    // fieldParts[0] is SRC_FIELD
+                    // fieldParts[1] is DST_FIELD (whindex)
+                    // fieldParts[2] is DELETE_SRC
+                    String[] fieldParts = field.split(SRC_DST_DEL_DELIMITER);
+                    String srcField = fieldParts[0];
+                    String whindexField = fieldParts[1];
+
+                    whindexFieldDefinitions.putAll(whindexField, Arrays.asList(curValues));
+
+                    // Only add the SRC_FIELD to the overloaded list if DELETE_SRC is true
+                    if (fieldParts.length > 2 && Boolean.parseBoolean(fieldParts[2])) {
+                        overloadedFields.add(srcField);
+                    }
+                }
+            }
+        }
+
+        /**
+         * Given a "{@code RULE}", return a {@code Multimap<String, String>} of whindex fields ("{@code DST_FIELD}") mapped to the values specified by the
+         * {@code RULE}.
+         *
+         * @return the mapping of whindex fields to values.
+         */
+        public Multimap<String,String> getWhindexFieldDefinitions() {
+            return whindexFieldDefinitions;
+        }
+
+        /**
+         * {@code OverloadedWhindexField}s are source fields ("{@code SRC_FIELD}") that become redundant once whindex entries are generated. They are marked to
+         * be removed.
+         *
+         * @return all {@code OverloadedWhindexField}s parsed from the {@code .rules}.
+         */
+        public Set<String> getOverloadedFields() {
+            return overloadedFields;
+        }
+
+        /**
+         * @param field
+         *            the field to check.
+         * @return {@code true} if {@code field} is a whindex field.
+         */
+        public boolean isWhindexField(String field) {
+            return whindexFieldDefinitions.containsKey(field);
+        }
+
+        // todo: discuss what this actually needs to do
+        public Multimap<String,NormalizedContentInterface> normalize(Multimap<String,String> fields) {
+            Multimap<String,NormalizedContentInterface> eventFields = HashMultimap.create();
+            for (Map.Entry<String,String> field : fields.entries()) {
+                eventFields.put(field.getKey(), new NormalizedFieldAndValue(field.getKey(), field.getValue()));
+            }
+            return normalizeMap(eventFields);
+        }
+
+        // todo: discuss what this actually needs to do
```

Comment:
This method is intended to take the input map of fields and values, and is to apply the whindex configuration.  This means creating the whindex fields per the configuration, dropping the source fields (of overloaded), etc.  This is intended to be business logic behind the creating of whindex fields.

See the comment above as to the fields that this method needs to create and fields this method needs to remove.  The map returned is a modification of the eventFields map passed in.